### PR TITLE
refactor: ScalarVisitor refactor push_down_prewhere

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

ScalarVisitor refactor push_down_prewhere

- Fixes #[Link the issue here]

## Tests

- [x] No Test  This pr is refactor code. If the result of explain is keep the origin, PR is ok.

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15215)

### CODE REVIEW

File: src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_prewhere.rs [hunk 1]
```diff
@@ -58,44 +63,65 @@ impl RulePushDownPrewhere {
...
             }
         }
-        Ok(())
+
+        let mut column_visitor = ColumnVisitor {
+            table_index,
+            schema: schema.clone(),
+            columns: ColumnSet::new(),
+        };
+        // WindowFunc, SubqueryExpr and AggregateFunction will not appear in Filter-LogicalGet
```
Comment (TCeason) [inline src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_prewhere.rs:122]: Maybe UDFCall can not appear in Filter-LogicalGet? But UDFLambdaCall can?

Comment (xudong963) [reply src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_prewhere.rs:113]: What's `Filter-LogicalGet` ?

Comment (TCeason) [reply src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_prewhere.rs:113]: I think it's Filter logical op? cc @b41sh

Comment (TCeason) [reply src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_prewhere.rs:113]: I think it's Filter logical op? cc @b41sh

Comment (b41sh) [reply src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_prewhere.rs:113]: aggregate and window function can't appear in filter, binder phase will check it.

Comment (TCeason) [reply src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_prewhere.rs]: > aggregate and window function can't appear in filter, binder phase will check it.

Yes. https://github.com/datafuselabs/databend/blob/b8f445bd59315d4d80f6306d515e4a11c29e0692/src/query/sql/src/planner/binder/select.rs#L468

**Review by b41sh — APPROVED** [review]

**Review by andylokandy — APPROVED** [review]

**Pull request merged**